### PR TITLE
Support MySQL/PostgreSQL cluster mode

### LINKED ISSUE #9762: MySQL/PostgreSQL monitor should be able to support cluster mode scenario
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/skywalking/issues?q=is%3Aissue) and found no similar feature requirement.

### Description

In recent pull requests from @yswdqz we can monitor metrics from MySQL and PostgreSQL, but currently we assume that every MySQL/PostgreSQL server only has one instance, and we use that instance as the service name, (refer to our demo environment http://demo.skywalking.apache.org/mySQL)

image

However, a MySQL/PostgreSQL server should be able to have multiple instances and form a cluster, in that case, we should have aggregated metrics for those instances.

### Use case

When there are multiple instances/nodes for one MySQL/PostgreSQL server, users can have an aggregated overview of the database server.

### Related issues

_No response_

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by yswdqz**
- [x] If this is non-trivial feature, paste the links/URLs to the design doc.
- [x] Update the documentation to include this new feature.
- [x] Tests(including UT, IT, E2E) are added to verify the new feature.
- [x] If it's UI related, attach the screenshots below.

- [x] If this pull request closes/resolves/fixes an existing issue, replace the issue number. Closes #9762 .
- [x] Update the [`CHANGES` log](https://github.com/apache/skywalking/blob/master/docs/en/changes/changes.md).

ScreenShots:
MySQL:
image
image
image
PostgreSQL:
OJ 2)M_W{ZF%OU8D{O62%5R
)TU2(JMLQX Y%%IWBET11ZA

_MYM@%6 {(5B0(7 57UPK

### CODE REVIEW

**Comment by wu-sheng** [comment]
Please fix conflicts and make CI running :)

File: docs/en/changes/changes.md [hunk 1]
```diff
@@ -83,6 +83,7 @@
 * Support `sampledTrace` in LAL.
 * Support multiple rules with different names under the same layer of LAL script.
 * (Optimization) Reduce the buffer size(queue) of MAL(only) metric streams. Set L1 queue size as 1/20, L2 queue size as 1/2.
+* Support MySQL/PostgreSQL cluster mode.
```
Comment (wu-sheng) [inline docs/en/changes/changes.md:86]:
```suggestion
* Support monitoring MySQL/PostgreSQL in the cluster mode.
```

MySQL and PG could be OAP server storage. We need to make this clear.

File: docs/en/setup/backend/backend-mysql-monitoring.md [hunk 2]
```diff
@@ -14,7 +14,8 @@ SkyWalking leverages prometheus/mysqld_exporter for collecting metrics data. It
 3. Config SkyWalking [OpenTelemetry receiver](opentelemetry-receiver.md).
 
 ### MySQL Monitoring
-MySQL monitoring provides monitoring of the status and resources of the MySQL server. MySQL server is cataloged as a `Layer: MYSQL` `Service` in OAP.
+MySQL monitoring provides monitoring of the status and resources of the MySQL server. MySQL cluster is cataloged as a `Layer: MYSQL` `Service` in OAP.
+A MySQL Server is cataloged as a `Instance` in OAP.
```
Comment (wu-sheng) [inline docs/en/setup/backend/backend-mysql-monitoring.md:18]:
```suggestion
Each MySQL server is cataloged as an `Instance` in OAP.
```

File: docs/en/setup/backend/backend-postgresql-monitoring.md [hunk 3]
```diff
@@ -64,8 +64,8 @@ SkyWalking leverages [fluentbit](https://fluentbit.io/) or other log agents for
 3. Config PostgreSQL to enable slow log. [Example](../../../../test/e2e-v2/cases/postgresql/postgres-exporter/postgresql.conf).
 
 ### Slow SQL Monitoring
-Slow SQL monitoring provides monitoring of the slow SQL statements of the PostgreSQL server. PostgreSQL server is cataloged as a `Layer: POSTGRESQL` `Service` in OAP.
-
+Slow SQL monitoring provides monitoring of the slow SQL statements of the PostgreSQL server. PostgreSQL Cluster is cataloged as a `Layer: POSTGRESQL` `Service` in OAP.
+A PostgreSQL Server is cataloged as a `Instance` in OAP.
```
Comment (wu-sheng) [inline docs/en/setup/backend/backend-postgresql-monitoring.md:68]: Same as above.

**Review by wu-sheng — COMMENTED** [review]
Docs nits.

**Comment by wu-sheng** [comment]
Is PG e2e having some issue? It failed twice.

**Comment by yswdqz** [comment]
> Is PG e2e having some issue? It failed twice.

Yes, I am trying to fix it now.

**Review by wu-sheng — APPROVED** [review]
LGTM once e2e passed.

**Comment by wu-sheng** [comment]
Anything should be changed on the showcase? Or it is compatible with previous setup?

File: test/e2e-v2/cases/postgresql/postgres-exporter/docker-compose.yml [hunk 4]
```diff
@@ -43,14 +43,44 @@ services:
...
+      - TZ=Asia/Shanghai
+      - POSTGRES_PASSWORD=123456
+      - POSTGRES_DB=skywalking
+    ports:
+      - 5432
+    command: postgres -c config_file=/etc/postgresql.conf
+    volumes:
+      - ./postgresql.conf:/etc/postgresql.conf
+    networks:
+      - e2e
```
[the remark this answers is not in the document]
Comment (yswdqz) [reply test/e2e-v2/cases/postgresql/postgres-exporter/docker-compose.yml:84]: They are not the same, such as mounting volumes

[the remark this answers is not in the document]
Comment (kezhenxu94) [reply test/e2e-v2/cases/postgresql/postgres-exporter/docker-compose.yml:84]: At least, `postgres_2` and `postgres_3` are the same

[the remark this answers is not in the document]
Comment (yswdqz) [reply test/e2e-v2/cases/postgresql/postgres-exporter/docker-compose.yml:84]: Got it.

**Comment by wu-sheng** [comment]
so11y seems to have failed several times. I am not sure whether relevant or not.

**Comment by yswdqz** [comment]
> so11y seems to have failed several times. I am not sure whether relevant or not.

I only edit about oap is binary.xml (because of my last PR), I think it should not relevant.

**Review by wu-sheng — APPROVED** [review]

**Pull request merged**